frontend/src/metabase/actions/components/ActionViz/Action.tsx

```diff
@@ -160,10 +174,32 @@ function ActionFn(props: ActionProps) {
   return <ConnectedActionComponent {...props} />;
 }
 
+function getErrorTooltip({
+  hasActionAssigned,
+  hasActionsEnabled,
+  error,
+}: {
+  hasActionAssigned: boolean;
+  hasActionsEnabled: boolean;
+  error?: GenericErrorResponse;
```

Comment:
nit: I'd type `error` as `unknown` because it can be an exception and not the error from the backend. `getResponseErrorMessage` decides whether the error is something that looks like a backend error response